mucommander-core/src/main/java/com/mucommander/ui/main/MainFrame.java

```diff
@@ -114,24 +119,58 @@ public class MainFrame extends JFrame implements LocationListener {
      */
     private void setWindowIcon() {
         // TODO: this code should probably be moved to the desktop API
-
         // - Mac OS X completely ignores calls to #setIconImage/setIconImages, no need to waste time
-        if(OsFamily.MAC_OS.isCurrent())
-            return;
+        if(!OsFamily.MAC_OS.isCurrent()) {
+
+            List<Image> icons = new ArrayList<>();
+
+            // Start by adding a 16x16 image with 1-bit transparency, any OS should support that.
+            icons.add(IconManager.getIcon(IconManager.MUCOMMANDER_ICON_SET, "icon16_8.png").getImage());
+
+            // Add PNG 24 images (8-bit transparency)
+            icons.add(IconManager.getIcon(IconManager.MUCOMMANDER_ICON_SET, "icon16_24.png").getImage());
+            icons.add(IconManager.getIcon(IconManager.MUCOMMANDER_ICON_SET, "icon32_24.png").getImage());
+            icons.add(IconManager.getIcon(IconManager.MUCOMMANDER_ICON_SET, "icon48_24.png").getImage());
+            icons.add(IconManager.getIcon(IconManager.MUCOMMANDER_ICON_SET, "icon128_24.png").getImage());
+            icons.add(IconManager.getIcon(IconManager.MUCOMMANDER_ICON_SET, "icon256_24.png").getImage());
+
+            setIconImages(icons);
+        }
 
-        java.util.List<Image> icons = new Vector<Image>();
+        JobsManager.getInstance().addJobListener(new JobListener() {
```

Comment:
Is it a good place to add such a listener?

Reply:
well, generally I don't see the benefit of using multiple frames now that we have tabs and the ability to execute operations in the background but as this option of having multiple frames still exists, I'd say we shouldn't add a listener per frame. wouldn't it make more sense to integrate it into the `com.mucommander.ui.notifier` package?